tests/test_api_items.py

```diff
@@ -0,0 +1,103 @@
+        item_name = json.dumps({"name": "something to do"})
+        resp = self.app.post("/api/v1/bucketlists/1/items", data=item_name,
+                             headers=self.header, content_type=self.mime_type)
+        resp_data = json.loads(resp.data)
+        new = Item.query.count()
+        self.assertEqual(1, new-old)
+        self.assertEqual(201, resp.status_code)
+        self.assertEqual("item has been added to the bucketlist",
+                         resp_data["message"])
+
+    def test_add_item_wrong_inputs(self):
+        """ test that blank spaces, empty arguments not accepted when adding
+        item"""
+        blank = json.dumps({})
+        empty = json.dumps({"name": ""})
+        space = json.dumps({"name": " "})
+
+        resp_blank = self.app.post("/api/v1/bucketlists/1/items",
+                                   data=blank, headers=self.header,
+                                   content_type=self.mime_type)
+        resp_empty = self.app.post("/api/v1/bucketlists/1/items",
+                                   data=empty, headers=self.header,
+                                   content_type=self.mime_type)
+        resp_space = self.app.post("/api/v1/bucketlists/1/items",
+                                   data=space, headers=self.header,
+                                   content_type=self.mime_type)
+        resp_blank_data = json.loads(resp_blank.data)
+        resp_empty_data = json.loads(resp_empty.data)
+        resp_space_data = json.loads(resp_space.data)
+
+        self.assertEqual(1, Item.query.count())
+        self.assertListEqual([400, 400, 400],
+                             [resp_blank.status_code, resp_empty.status_code,
+                              resp_space.status_code])
+        self.assertIn("name required", resp_blank_data["message"]["name"])
+        self.assertEqual("no blank fields allowed", resp_empty_data["message"])
+        self.assertEqual("name is invalid", resp_space_data["message"])
+
+    def test_delete_item(self):
+        """ test that item is deleted """
+        old = Item.query.count()
+        resp = self.app.delete("/api/v1/bucketlists/1/items/1",
+                               headers=self.header)
+        resp_data = json.loads(resp.data)
+        new = Item.query.count()
+        self.assertEqual(1, old-new)
+        self.assertEqual(200, resp.status_code)
+        self.assertIn("deleted successfully", resp_data["message"])
+
+        noitem_resp = self.app.delete("/api/v1/bucketlists/1/items/1",
+                                      headers=self.header)
+        noitem_resp_data = json.loads(noitem_resp.data)
+        self.assertEqual(404, noitem_resp.status_code)
+        self.assertIn("item not found", noitem_resp_data["message"])
+
+    def test_update_item(self):
+        """ test that item can be updated and that at least
+        one argument must be provided"""
+        valid_data = json.dumps({"name": "updated list", "status": "true"})
```

Comment:
This part could be abstracted using a base class where you define all the necessary properties and attributes and then inherit that class.  Peep this example and let me know in case something does not make sense :https://github.com/andela-kndegwa/CP3/blob/ch-write-tests/zuhura/api/tests/utils.py